packages/image_picker/image_picker_ios/ios/Classes/FLTPHPickerSaveImageToPathOperation.m

```diff
@@ -88,25 +88,20 @@ - (void)start {
   if (@available(iOS 14, *)) {
     [self setExecuting:YES];
 
-    if ([self.result.itemProvider hasItemConformingToTypeIdentifier:UTTypeWebP.identifier]) {
+    // This supports uniform types that conform to UTTypeImage.
+    // This includes UTTypeHEIC, UTTypeHEIF, UTTypeLivePhoto, UTTypeICO, UTTypeICNS, UTTypePNG
+    // UTTypeGIF, UTTypeJPEG, UTTypeWebP, UTTypeTIFF, UTTypeBMP, UTTypeSVG, UTTypeRAWImage
+    if ([self.result.itemProvider hasItemConformingToTypeIdentifier:UTTypeImage.identifier]) {
       [self.result.itemProvider
-          loadDataRepresentationForTypeIdentifier:UTTypeWebP.identifier
+          loadDataRepresentationForTypeIdentifier:UTTypeImage.identifier
                                 completionHandler:^(NSData *_Nullable data,
                                                     NSError *_Nullable error) {
-                                  UIImage *image = [[UIImage alloc] initWithData:data];
-                                  [self processImage:image];
+                                  if (error == nil && data != nil) {
+                                    UIImage *image = [[UIImage alloc] initWithData:data];
+                                    [self processImage:image];
+                                  }
```

Comment:
Note we have an outstanding issue where if there is an error nothing happens and the result completion block isn't called, and you get a `PlatformException` when you try again because the old request is still in-flight https://github.com/flutter/flutter/issues/82602.

This PR doesn't make that any worse, but it would be good to at least add some logging as a hint about what went wrong until that can be resolved:
```suggestion
                                  } else {
                                    os_log_error(OS_LOG_DEFAULT, "Could not process image: %@", error);
                                  }  
```

Reply:
Will do. I had noticed in [previous PR](https://github.com/flutter/plugins/pull/4448#discussion_r768872859) they had contemplated fixing same issue with errors not being handled but decided to handle in a different issue: https://github.com/flutter/flutter/issues/95268